tests/Test/Xrefcheck/TooManyRequestsSpec.hs

```diff
@@ -0,0 +1,59 @@
+{- SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+ -
+ - SPDX-License-Identifier: MPL-2.0
+ -}
+
+module Test.Xrefcheck.TooManyRequestsSpec where
+
+import Universum
+
+import Control.Concurrent (forkIO, killThread)
+import qualified Control.Exception as E
+import qualified Data.CaseInsensitive as CI
+import qualified Data.Map as M
+import Network.HTTP.Types (Status (..), ok200, serviceUnavailable503, tooManyRequests429)
+import Network.HTTP.Types.Header (hRetryAfter)
+import Test.Hspec (Spec, describe, it, shouldBe)
+import Web.Firefly (ToResponse (toResponse), route, run)
+
+import Xrefcheck.Config
+import Xrefcheck.Core
+import Xrefcheck.Verify
+
+spec :: Spec
```

Comment:
I've considered the latter test case but stepped back from implementing it at that time because I thought waiting additional 20 seconds for the test suite to complete (which, I believe, takes up more time than the entire set of test suites minus the 429 test case) would be a bit long. I, nevertheless, will implement both & see!

I will also look up if setting the retry-after value to 0 is valid or not, because if it is, then we won't have to wait a minimum of 20 seconds for that test case to finish running.

Reply:
Ah good point!

An alternative would be to test this manually for now, and write an automated test only when `maxRetries` becomes configurable (because then we'll be able to set `maxRetries` to 2 or something like that).

On that topic: I created an issue for making `maxRetries` configurable, #104.

I'll let you decide what to do regarding test (2). If you do decide to do it later, please add a note about that to #104 so we don't forget about it.